<?xml version="1.0"?>
<launch>
  <arg name="monitor_namespace"/>
  <arg name="monitored_namespace" default=""/>
  <arg name="robot_description" default="robot_description"/>
  <arg name="discrete_plugin" default="BulletDiscreteBVHManager"/>
  <arg name="continuous_plugin" default="BulletCastBVHManager"/>
  <arg name="monitor_links" default=""/>
  <arg name="disabled_links" default=""/>
  <arg name="publish_environment" default="false"/>
  <arg name="publish_markers" default="false"/>

  <!--
  FIRST = 0,   /**< Return at first contact for any pair of objects */
  CLOSEST = 1, /**< Return the global minimum for a pair of objects */
  ALL = 2,     /**< Return all contacts for a pair of objects */
  LIMITED = 3  /**< Return limited set of contacts for a pair of objects */
  -->
  <arg name="contact_test_type" default="2"/>
  <arg name="contact_distance"/>

  <node pkg="tesseract_monitoring" type="tesseract_monitoring_contacts_node" name="tesseract_monitoring_contacts" output="screen">
    <param name="robot_description" type="string" value="$(arg robot_description)"/>
    <param name="discrete_plugin" type="string" value="$(arg discrete_plugin)"/>
    <param name="continuous_plugin" type="string" value="$(arg continuous_plugin)"/>
    <param name="contact_distance" value="$(arg contact_distance)"/>
    <param name="monitor_links" value="$(arg monitor_links)"/>
    <param name="disabled_links" value="$(arg disabled_links)"/>
    <param name="contact_test_type" value="$(arg contact_test_type)"/>
    <param name="monitor_namespace" type="string" value="$(arg monitor_namespace)"/>
    <param name="monitored_namespace" type="string" value="$(arg monitored_namespace)"/>
    <param name="publish_environment" type="bool" value="$(arg publish_environment)"/>
    <param name="publish_markers" type="bool" value="$(arg publish_markers)"/>
  </node>

</launch>
